packages/bee-q/src/components/breadcrumb/bq-breadcrumb.tsx

```diff
@@ -0,0 +1,135 @@
+
+  // Prop lifecycle events
+  // =======================
+
+  // Events section
+  // Requires JSDocs for public API documentation
+  // ==============================================
+
+  /** Handler to be called when `bq-breadcrumb-item` item loses focus. */
+  @Event() bqBreadcrumbBlur: EventEmitter<HTMLBqBreadcrumbItemElement>;
+
+  /** Handler to be called when `bq-breadcrumb-item` item gets focus. */
+  @Event() bqBreadcrumbFocus: EventEmitter<HTMLBqBreadcrumbItemElement>;
+
+  /** Handler to be called when `bq-breadcrumb-item` is selected (on click/enter press). */
+  @Event() bqBreadcrumbClick: EventEmitter<HTMLBqBreadcrumbItemElement>;
+
+  // Component lifecycle events
+  // Ordered by their natural call order
+  // =====================================
+
+  // Listeners
+  // ==============
+
+  @Listen('bqBlur', { passive: true })
+  onBlur(event: CustomEvent<HTMLElement>) {
+    if (isHTMLElement(event.detail, 'bq-breadcrumb-item')) this.bqBreadcrumbBlur.emit(event.detail);
+  }
+
+  @Listen('bqFocus', { passive: true })
+  onFocus(event: CustomEvent<HTMLElement>) {
+    if (isHTMLElement(event.detail, 'bq-breadcrumb-item')) this.bqBreadcrumbFocus.emit(event.detail);
+  }
+
+  @Listen('bqClick', { passive: true })
+  onClick(event: CustomEvent<HTMLElement>) {
+    if (isHTMLElement(event.detail, 'bq-breadcrumb-item')) this.bqBreadcrumbClick.emit(event.detail);
+  }
+
+  // Public methods API
+  // These methods are exposed on the host element.
+  // Always use two lines.
+  // Public Methods must be async.
+  // Requires JSDocs for public API documentation.
+  // ===============================================
+
+  // Local methods
+  // Internal business logic.
+  // These methods cannot be called from the host element.
+  // =======================================================
+
+  private setSeparator = (): void => {
+    this.breadcrumbItems.forEach((item, index, arr) => {
+      item.isLastItem = index === arr.length - 1;
+      !item.isLastItem && item.append(this.getSeparatorElem());
+    });
+  };
+
+  /**
+   * clone original element and add slot attr
+   * @returns cloned separator element
+   */
+  private getSeparatorElem = (): HTMLElement => {
+    const separator = this.separatorFromSlot || this.defaultSeparator;
+    const clone = separator.cloneNode(true) as HTMLElement;
+    clone.slot = 'separator';
+
+    return clone;
+  };
+
+  private get breadcrumbItems(): HTMLBqBreadcrumbItemElement[] {
+    return Array.from(this.el.querySelectorAll('bq-breadcrumb-item'));
+  }
+
+  private get separatorFromSlot() {
+    return this.spanElem
+      .querySelector<HTMLSlotElement>('slot[name="separator"]')
+      .assignedElements({ flatten: true })[0] as HTMLElement;
+  }
+
+  private get defaultSeparator(): HTMLElement {
+    const newNode = document.createElement('span');
+    newNode.classList.add('flex', 'w-3', 'items-center', 'justify-center');
+    newNode.textContent = '/';
+
+    return newNode;
+  }
```

Comment:
I just tested adding a custom separator and this is what happened:

![CleanShot 2023-07-12 at 20 02 29](https://github.com/Endava/bee-q/assets/328492/87a4cc4b-0996-4682-adaa-732e8ef28df4)

Also, if the separator change, it is not reflected, eg. if I change the custom separator icon size or the name.

One more thing, it won't be easier if we just append the separator after each breadcrumb item (except the last one) instead of adding it inside a `bq-breadcrumb-item` slot?

Reply:
I see your point. From integration it works as intended, but in case of changing the separator dynamically or manually it doesn't work. I'll fix it.
I think that the separator element is relevant in the same context as breadcrumb item. Let s say we remove an item, then we want to remove the inner text and also the separator, together. That's why the separator is placed inside the item. Thoughts?

Reply:
Yes, it makes sense regarding where to attach the separator. Let's keep it that way then 👍🏼